Lib/glyphsLib/filters/eraseOpenCorners.py

```diff
@@ -57,7 +57,7 @@ def endPath(self):
             # of the next segment both on the right side of the line?
             # (see discussion at https://github.com/googlefonts/glyphsLib/pull/663)
             pt1 = segs[ix - 1][0]
-            pt2 = segs[next_ix][-1]
+            pt2 = segs[next_ix][1]
```

Comment:
This indeed fixes the Sofia Sans `kgreenlandic` problem :) I don't understand the logic, though, is the comment above up to date?

Reply:
Yes. The comment is fine. But the code before only worked for line segments, not curves. This is an example of the problem:

![](https://user-images.githubusercontent.com/106728/134326400-3307cea7-0368-4441-9ff5-8ba3e4e61fd0.png)

We know something is an open corner if both "points" are on the same side of the yellow line. But which points? Previously we were looking at the *ends* of the surrounding segments ( (444,362) and (432,213) ). But if you extend the yellow line, you'll find that (432,213) is actually on the other side of it! We should instead be checking the outgoing handle (353,296).

Hence, point 2 is either the outgoing handle for cubics or the end of the line segments for lines - in either case, `curve[1]`, not `curve[-1]`.

Reply:
Sorry, the comment says "last point", and you're right, that's outdated.

Reply:
Oh, this reveals another bug...